### PR TITLE
Round 2 of setup_run_environment cleanup for FW_SEARCH_PATH FHICL_FILE_PATH and CET_PLUGIN_PATH

### PR DESCRIPTION
**Pull request by marcmengel**
Trying again with a cleaner branch.

### CODE REVIEW

File: packages/cetlib-except/package.py [hunk 1]
```diff
@@ -61,3 +61,7 @@ def setup_build_environment(self, env):
         env.prepend_path("PATH", os.path.join(self.build_directory, "bin"))
         # Cleanup.
         sanitize_environments(env, "PATH")
+
+    def setup_run_environment(self, run_env):
+        run_env.prepend_path("ROOT_INCLUDE_PATH", self.prefix.include)
+        run_env.append_path("CET_PLUGIN_PATH", self.prefix.lib)
```
Comment (knoepfel) [inline packages/cetlib-except/package.py]: `cetlib-except` should not need to affect these environment variables.  I'd remove `setup_run_environment` here.

File: packages/canvas/package.py [hunk 2]
```diff
@@ -76,3 +76,7 @@ def setup_build_environment(self, env):
         env.prepend_path("PATH", os.path.join(prefix, "bin"))
         # Cleanup.
         sanitize_environments(env, "PATH")
+
+    def setup_run_environment(self, run_env):
+        run_env.prepend_path("ROOT_INCLUDE_PATH", self.prefix.include)
+        run_env.append_path("CET_PLUGIN_PATH", self.prefix.lib)
```
Comment (knoepfel) [inline packages/canvas/package.py]: `canvas` should not need to affect these variables.  I'd remove `setup_run_environment` here.

File: packages/canvas-root-io/package.py [hunk 3]
```diff
@@ -96,6 +96,8 @@ def setup_run_environment(self, env):
         ):
             env.prepend_path("ROOT_INCLUDE_PATH", str(self.spec[d.name].prefix.include))
         env.prepend_path("ROOT_INCLUDE_PATH", prefix.include)
+        env.append_path("CET_PLUGIN_PATH", self.prefix.lib)
```
Comment (knoepfel) [inline packages/canvas-root-io/package.py]: `canvas-root-io` does not provide any plugins, so I'd remove this.

File: packages/fhicl-cpp/package.py [hunk 4]
```diff
@@ -78,8 +78,11 @@ def setup_build_environment(self, env):
         sanitize_environments(env, "PATH")
 
     def setup_run_environment(self, env):
-        bindir = self.prefix.bin
+        env.prepend_path("PATH", self.prefix.bin)
+        env.prepend_path("ROOT_INCLUDE_PATH", self.prefix.include)
+        env.append_path("CET_PLUGIN_PATH", self.prefix.lib)
```
Comment (knoepfel) [inline packages/fhicl-cpp/package.py]: `fhicl-cpp` should not need to adjust `ROOT_INCLUDE_PATH` and `CET_PLUGIN_PATH`--I'd remove these two lines.

File: packages/hep-concurrency/package.py [hunk 5]
```diff
@@ -71,3 +71,7 @@ def setup_build_environment(self, env):
         env.prepend_path("PATH", os.path.join(prefix, "bin"))
         # Cleanup.
         sanitize_environments(env, "PATH")
+
+    def setup_run_environment(self, run_env):
+        run_env.prepend_path("ROOT_INCLUDE_PATH", self.prefix.include)
+        run_env.append_path("CET_PLUGIN_PATH", self.prefix.lib)
```
Comment (knoepfel) [inline packages/hep-concurrency/package.py]: `hep-concurrency` should not need to adjust these environment variables.  I'd remove `setup_run_environment`.

File: packages/nusimdata/package.py [hunk 6]
```diff
@@ -106,12 +106,15 @@ def setup_run_environment(self, run_env):
...
+            cover="nodes",
+            order="post",
+            deptype=("link"),
+            direction="children",
         ):
             run_env.prepend_path("ROOT_INCLUDE_PATH", str(self.spec[d.name].prefix.include))
         run_env.prepend_path("ROOT_INCLUDE_PATH", self.prefix.include)
-        # Perl modules.
-        run_env.prepend_path("PERL5LIB", os.path.join(self.prefix, "perllib"))
+        run_env.prepend_path("FHICL_FILE_PATH", self.prefix.fcl)
```
Comment (knoepfel) [inline packages/nusimdata/package.py]: `nusimdata` provides no FHiCL files--I'd remove this line.

File: packages/larsoft/package.py [hunk 7]
```diff
@@ -137,14 +137,11 @@ def setup_dependent_build_environment(self, spack_env, dspec):
...
-    def setup_dependent_run_environment(self, run_env, dspec):
-        run_env.prepend_path("PATH", self.prefix.bin)
-        run_env.prepend_path("ROOT_INCLUDE_PATH", self.prefix.include)
+    def setup_run_environment(self, run_env):
         run_env.prepend_path("PATH", self.prefix.bin)
         run_env.prepend_path("ROOT_INCLUDE_PATH", self.prefix.include)
         run_env.append_path("FHICL_FILE_PATH", "{0}/fcl".format(self.prefix))
-        run_env.append_path("FW_SEARCH_PATH", "{0}/fw".format(self.prefix))
-        sanitize_environments(run_env)
+        run_env.append_path("CET_PLUGIN_PATH", self.prefix.lib)
```
Comment (knoepfel) [inline packages/larsoft/package.py]: As `larsoft` is an umbrella product, it provides no FHiCL files or C++ code.  I'd therefore keep only the adjustment for `PATH`.

File: packages/gallery/package.py [hunk 8]
```diff
@@ -75,3 +75,11 @@ def setup_dependent_build_environment(self, env, dependent_spec):
         env.prepend_path("CET_PLUGIN_PATH", prefix.lib)
         # Cleaup.
         sanitize_environments(env, "CET_PLUGIN_PATH")
+
+    def setup_run_environment(self, run_env):
+        run_env.prepend_path("PATH", self.prefix.bin)
+        run_env.prepend_path("ROOT_INCLUDE_PATH", self.prefix.include)
+        run_env.append_path("FHICL_FILE_PATH", "{0}/fcl".format(self.prefix))
+        run_env.append_path("FW_SEARCH_PATH", "{0}/fw".format(self.prefix))
+        run_env.append_path("CET_PLUGIN_PATH", self.prefix.lib)
```
Comment (knoepfel) [inline packages/gallery/package.py]: `gallery` is just a library--it provides no executables, FHiCL files or plugins.  I think the only adjustment to include is the one for `ROOT_INCLUDE_PATH`.

**Review by knoepfel — CHANGES_REQUESTED** [review]
Much better, @marcmengel.  I flagged several issues, but the vast majority of the improvements can go in without modification.

Thanks!

File: packages/cetlib-except/package.py (hunk 1 above)
Comment (marcmengel) [reply packages/cetlib-except/package.py]: Done.

File: packages/canvas/package.py (hunk 2 above)
Comment (marcmengel) [reply packages/canvas/package.py]: Done

File: packages/canvas-root-io/package.py (hunk 3 above)
Comment (marcmengel) [reply packages/canvas-root-io/package.py]: Done.

File: packages/gallery/package.py (hunk 8 above)
Comment (marcmengel) [reply packages/gallery/package.py]: Done

File: packages/larsoft/package.py (hunk 7 above)
Comment (marcmengel) [reply packages/larsoft/package.py]: Done.

File: packages/nusimdata/package.py (hunk 6 above)
Comment (marcmengel) [reply packages/nusimdata/package.py]: Done.

File: packages/hep-concurrency/package.py (hunk 5 above)
Comment (marcmengel) [reply packages/hep-concurrency/package.py]: Done

**Review by knoepfel — APPROVED** [review]

**Comment by knoepfel** [comment]
@marcmengel, thanks.  This is a nice change.

**Pull request merged**